content/docs/tasks.md

```diff
@@ -0,0 +1,178 @@
+
+```yaml
+tasks:
+  - name: genwww
+    kind: Pipe
+    spec:
+      pod: pipe-pod.yaml
+      pipe:
+        - file: /tmp/index.html
+          kind: ConfigMap
+          key: indexHtml
+```
+
+a pipe-task spec has two fields: a `pod` spec that will generate our `index.html` and a `pipe` spec that defines a list of files that will be persisted. Here we'll generate a `/tmp/index.html` file that we'll save as a `ConfigMap`. It can be referenced with the key `indexHtml` (more about that below). Pipe-task `spec.pod` field must reference a [core/v1 Pod](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#pod-v1-core) template. However, there are limitations. Reasons for that are explained in the [corresponding KEP](https://github.com/kudobuilder/kudo/blob/master/keps/0017-pipe-tasks.md). In a nutshell:
+
+- A pipe-pod should generate artifacts in a [initContainer](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/)
+- It has to define and mount one [emptyDir volume](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir) where generated files are stored
+
+Let's take a look at `templates/pipe-pod.yaml`:
+
+```yaml
+apiVersion: v1
+kind: Pod
+spec:
+  volumes:
+  - name: shared-data
+    emptyDir: {}
+  initContainers:
+    - name: init
+      image: busybox
+      command: [ "/bin/sh", "-c" ]
+      args:
+        - wget -O /tmp/index.html 'http://cowsay.morecode.org/say?message=Good+things+come+when+you+least+expect+them&format=html'
+      volumeMounts:
+        - name: shared-data
+          mountPath: /tmp
+```
+
+Here we use an online [cowsay-generator](http://cowsay.morecode.org) API to generate and download an `index.html` file and save in the mounted volume under `/tmp/index.html` path.  
+
+Given the above pipe-pod and `genwww` task specification, KUDO will run the pipe-pod, wait for the successful generation of the `index.html`, copy it out and save it as a ConfigMap. Now, this file is ready to be used in our Nginx deployment spec. Let's extend the `templates/deployment.yaml` to mount the generated file:
+
+```yaml
+...
+spec:
+  containers:
+    - name: nginx
+      image: nginx:1.7.9
+      ports:
+        - containerPort: 80
+      volumeMounts:
+        - name: www
+          mountPath: /usr/share/nginx/html/
+  volumes:
+    - name: www
+      configMap:
+        name: {{ .Pipes.indexHtml }}
+```
+
+We successfully created a [mount volume from the ConfigMap](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#populate-a-volume-with-data-stored-in-a-configmap). Note the parametrized ` {{ .Pipes.indexHtml } }` configMap name. `.Pipes` keyword allows us to address pipe-artifacts and `indexHtml` is the key we defined in the pipe-task above.
```

Comment:
`{{ .Pipes.indexHtml } }` -> `{{ .Pipes.indexHtml }}`

Reply:
Nope 😂 vuepress breaks because of `}}` and escaping them doesn't help.